### DISCUSSION TITLE
Should operations be returning sql.ErrNoRows, if they return a pointer anyway?

### DISCUSSION THREAD

**Discussion by preslavrachev** [aarondl/sqlboiler#1237] [Q&A] [answered]
I believe a version of this question has been asked in the past. Basically, the premise is this - assuming that `One` and `All` would return a pointer and an error anyway, does it make sense for the error to include `sql.ErrNoRows`? In my view, we could simply check for the non-nillability of the value, and leave the error for something more dramatic (like, a DB connection failure).

I presume that it is impossible to add this check by default because of the backwards compatibility, but could this be done with a configuration option?

**Comment by stephenafamo** [comment, marked as answer]
If asking for a single record, it is an error if none exists, but if asking for a list, you can get an empty list.

This is to match the behavior of the standard library. 

* `Query` and `QueryContext` do not return `sql.ErrNoRows`
* `QueryRow` and `QueryRowContext` return `sql.ErrNoRows`

So in SQLBoiler, `Find`, `One` return `sql.ErrNoRows` while `All` does not.
